### ISSUE TITLE
Asynchronous task

### ISSUE THREAD

**Issue by tingfeng-key** [mdsherry/clokwerk#23]
How to call asynchronous function

**Comment by mdsherry** [comment]
With some difficulty.

I recently had to create an async task, and this is the approach I took there:
```rust
// Fake values for the demonstration
let value = Arc::new(Mutex::new(vec![1,2,3,4]));
scheduler.every(1.days()).at("13:00").run(move || {
    // We've moved the outer Arc into the closure, and then clone it so we can move that into the async block.
    let value = value.clone();
    tokio::spawn(async {
        do_async_thing(value).await;
        do_other_async_thing().await;
    });
});
```

The main annoyance with a `spawn` based approach, apart from tying your code to a given executor, is that while the closure passed to `run` has a fairly well-defined lifetime bounded by that of the `scheduler`, anything passed to `tokio::spawn`, etc. has to have a `'static` lifetime.

I'll add a `run_pending_async` method if I can make it work nicely.

**Comment by tingfeng-key** [comment]
Thank you very much for your reply. This example can be run

**Comment by mdsherry** [comment]
Version 0.4, which is the current `master` branch but still unpublished, adds a new `AsyncScheduler`. You still need the `Arc` etc. because the lifetime of the Future needs to be independent of the closure, but you no longer need the `tokio::spawn`.

**Comment by dessalines** [comment]
This should probably be reopened until this is published.

**Comment by miqbalrr** [comment]
so when it will be published ?

**Comment by mdsherry** [comment]
It's published right now as https://crates.io/crates/clokwerk/0.4.0-rc1 . I'd like to know that the current design meets people's needs before I publish it as 0.4

**Comment by dlight** [comment]
@mdsherry is there any outstanding issue before releasing 0.4? Perhaps it's better to release this as is, and if issues are found maybe release a 0.5 later

I liked that it isn't tied to a specific executor (and I thought that timers generally require deep executor integration - do you avoid that by having a background thread spawn tasks or something?)

**Comment by luxalpa** [comment]
Trying to figure out how to use AsyncScheduler. Any examples / pointers?

**Comment by dlight** [comment]
When you go to the docs it goes to the latest stable release, you need to specifically ask for 0.4.0-rc1, and then you get this which has some examples

https://docs.rs/clokwerk/0.4.0-rc1/clokwerk/struct.AsyncScheduler.html

**Comment by dlight** [comment]
It would be really great to have 0.4 finally released, 0.4.0-rc1 was published on crates.io over a year ago (on the very same day 0.3.5 was out)
